src/pygeoprocessing/geoprocessing.py

```diff
@@ -1408,373 +1421,357 @@ def zonal_statistics(
-        of 'min' 'max' 'sum' 'count' and 'nodata_count'.  Example::
-
-            {0: {'min': 0,
-                 'max': 14,
-                 'sum': 42,
-                 'count': 8,
-                 'nodata_count': 1,
-                 'value_counts': {
-                     2: 5,
-                     4: 1,
-                     14: 2,
+        If `base_raster_path_bands` is a tuple, the return value is a nested
+        dictionary of stats for that raster band. Top-level keys are the
+        aggregate feature FIDs. Each nested FID dictionary then contains
+        statistics about that feature: 'min', 'max', 'sum', 'count',
+        'nodata_count', and optionally 'value_counts'. Example::
+            {
+                0: {
+                    'min': 0,
+                    'max': 14,
+                    'sum': 42,
+                    'count': 8,
+                    'nodata_count': 1,
+                    'value_counts': {
+                        2: 5,
+                        4: 1,
+                        14: 2
                     }
-                 }
+                }
             }
 
+        If `base_raster_path_bands` is a list of tuples, the return value is
+        a list of the nested dictionaries described above. Each dictionary in
+        the list contains the stats calculated for the corresponding raster
+        band in the `base_raster_path_bands` list.
+
+
     Raises:
         ValueError
-            if ``base_raster_path_band`` is incorrectly formatted.
+            if ``base_raster_path_band`` is incorrectly formatted, or if
+            not all of the input raster bands are aligned with each other
+
         RuntimeError
-            if the aggregate vector or layer cannot open.
+            if the aggregate vector or layer cannot be opened
 
     """
-    if not _is_raster_path_band_formatted(base_raster_path_band):
-        raise ValueError(
-            "`base_raster_path_band` not formatted as expected.  Expects "
-            "(path, band_index), received %s" % repr(base_raster_path_band))
+    # Check that the raster path/band input is formatted correctly
+    multi_raster_mode = isinstance(base_raster_path_bands, list)
+    if not multi_raster_mode:
+        base_raster_path_bands = [base_raster_path_bands]
+    for base_raster_path_band in base_raster_path_bands:
+        if not _is_raster_path_band_formatted(base_raster_path_band):
+            raise ValueError(
+                '`base_raster_path_band` not formatted as expected.  Expected '
+                f'(path, band_index), received {repr(base_raster_path_band)}')
+
+    # Check that all input rasters are aligned. This should hold true if they
+    # have the same projection, geotransform, and bounding box.
+    for attr in ['geotransform', 'bounding_box', 'projection_wkt']:
+        vals = set()
+        for path, _ in base_raster_path_bands:
+            raster_info = get_raster_info(path)
+            vals.add(str(raster_info[attr]))
```

Comment:
Oh that's an interesting point... I just used `str` to make the values hashable, so they could go in a set. I think in most cases, the rasters would have previously been run through `align_and_resize_raster_stack` or something equivalent, so I expect they'd be exactly the same. Maybe this could wait until we encounter a use case for it? That would also help inform what precision is needed.

Reply:
>Maybe this could wait until we encounter a use case for it?
Yes, that sounds good to me.

>I just used str to make the values hashable, so they could go in a set.
Aren't `float`s hashable already?

```
>>> a = set([])
>>> a.add(1.23456)
>>> a
{1.23456}
```